pytorch_lightning/utilities/cli.py

```diff
@@ -470,7 +476,7 @@ def __init__(
         save_config_multifile: bool = False,
         trainer_class: Union[Type[Trainer], Callable[..., Trainer]] = Trainer,
         trainer_defaults: Optional[Dict[str, Any]] = None,
-        seed_everything_default: Optional[int] = None,
+        seed_everything_default: Optional[Union[bool, uint32]] = True,
```

Comment:
Even though we have the deprecation path for Optional, I would keep the non-deprecated type here

```suggestion
        seed_everything_default: Union[bool, uint32] = True,
```